### PR TITLE
rescan: match against the raw script for an addr, not its datapush

### PR DESCRIPTION
**Pull request by Roasbeef**
In this commit, we update neutrino to match a recent change to the BIP
itself. Rather than looking for the push datas in a script, we'll
actually instead just look for the script itself. This is due to the
fact that we'll now index the entire script rather then each individual
push.

### CODE REVIEW

File: rescan.go [hunk 1]
```diff
@@ -712,8 +738,8 @@ type Rescan struct {
 }
 
 // NewRescan returns a rescan object that runs in another goroutine and has an
-// updatable filter. It returns the long-running rescan object, and a channel
-// which returns any error on termination of the rescan process.
+// updateable filter. It returns the long-running rescan object, and a channel
+// which returns any error on termination of the rescan prcess.
 func (s *ChainService) NewRescan(options ...RescanOption) Rescan {
```
Comment (aakselrod) [inline rescan.go]: This adds a new typo.

**Comment by aakselrod** [comment]
Verified tests succeed when run against https://github.com/Roasbeef/btcutil/pull/6 (with a `btcd` compiled with the patch).

**Comment by aakselrod** [comment]
Verified `lnd` tests pass against this PR and the `btcutil` PR when run with a `btcd` also built against the `btcutil` PR. Tested:

* `lnwallet`
* `chainntnfs`
* `chainview`

File: rescan.go (hunk 1 above)
Comment (Roasbeef) [reply rescan.go]: Fixed.

**Pull request merged**